habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/ValidatePythonCodeMojo.java

```diff
@@ -0,0 +1,80 @@
+package org.technologybrewery.habushu;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.technologybrewery.habushu.exec.PoetryCommandHelper;
+
+/**
+ * Leverages the pylint package to validate both source and test Python
+ * directories using Poetry's run command.
+ */
+@Mojo(name = "validate-python-code", defaultPhase = LifecyclePhase.PROCESS_CLASSES, requiresDependencyResolution = ResolutionScope.COMPILE)
+public class ValidatePythonCodeMojo extends AbstractHabushuMojo {
+
+    protected static final String PYLINT_PACKAGE = "pylint";
+
+    /**
+     * Specifies disabled checkers for pylint.
+     */
+    @Parameter(property = "habushu.pylintDisabledChecker", required = false, defaultValue = "C,R,W")
+    private String pylintDisabledChecker;
+
+    /**
+     * Specifies enabled checkers for pylint.
+     */
+    @Parameter(property = "habushu.pylintEnabledChecker", required = false, defaultValue = "E,F")
+    private String pylintEnabledChecker;
+
+    /**
+     * By default, build will stop if lint errors are found. Can be configured to true so that
+     * build will continue. 
+     */
+    @Parameter(property = "habushu.pylintExitZero", required = false, defaultValue = "false")
+    private boolean pylintExitZero;
+
+    @Override
+    public void doExecute() throws MojoExecutionException {
+        List<String> directoriesToValidate = new ArrayList<>();
+        if (this.sourceDirectory.exists()) {
+            directoriesToValidate.add(getCanonicalPathForFile(sourceDirectory));
```

Comment:
S: will this cover test clases?

Reply:
I'm opting out linting on the test classes because the tests are already executed by Python, hence Python would error out when the behave tests are executed.

Reply:
But wouldn't this possibly highlight a bad test?

Reply:
Yup, it would, lint would be able to identify bad code before tests are executed by Python. My only thoughts are that linting on test modules should have separate configurations from the source modules, so the Habushu configurations will need to be expanded. Expanding on this thought: there will be different requirements for checks in the source and test modules. For example, one of the errors I found when linting on tests is the overwriting function error, since a common function name for test steps is `step_impl`, but lint assumes you are overwriting the function with each step. Hence, we will need to disable the checker for overwriting function in the test or rename tests to be unique, but we do not want to disable this on the source module. 

But ultimately I think it's reasonable to lint on both source and test modules to give an earlier warning to users before the tests are executed.

Reply:
I'm pretty squarely in the camp that test code is first class code and should be treated just like source code.  Perhaps we should challenge some of the assumptions around overwriting et, al.  We might be able to find some useful tweaks that allow the same linting while also meeting our intent.